### PR TITLE
Sync `master` with `2201.0.3-stage-new`

### PR DESCRIPTION
**Pull request by IMS94**
## Purpose
$subject

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: .github/workflows/pull_request_windows_build.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ on:
       - stage-swan-lake
       - ballerina-[0-9]+.[0-9]+.x
       - 2201.[0-9]+.x
+      - 2201.0.3-stage-new
```
Comment (MaryamZi) [inline .github/workflows/pull_request_windows_build.yml:13]:
```suggestion
```

File: .github/workflows/pull_request_ubuntu_build.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ on:
       - ballerina-1.2.x
       - stage-swan-lake
       - 2201.[0-9]+.x
+      - 2201.0.3-stage-new
```
Comment (MaryamZi) [inline .github/workflows/pull_request_ubuntu_build.yml:13]:
```suggestion
```

Comment (HindujaB) [inline .github/workflows/pull_request_ubuntu_build.yml:13]: Shall we remove these lines as they are added to test the PR builds (in the other `yml` files as well)?

File: gradle.properties [hunk 3]
```diff
@@ -2,7 +2,7 @@ org.gradle.caching=true
 org.gradle.parallel=true
 org.gradle.jvmargs='-Dfile.encoding=UTF-8'
 org.gradle.workers.max=3
-version=2201.0.3-SNAPSHOT
+version=2201.0.3-rc1.6-SNAPSHOT
```
Comment (HindujaB) [inline gradle.properties:5]: We will need to change this after the release.

Comment (IMS94) [reply gradle.properties]: Updated to `2201.0.3-SNAPSHOT` for now

**Review by gimantha — APPROVED** [review]

**Pull request merged**